dali/operators/image/resize/tensor_resize_attr.cc

```diff
@@ -0,0 +1,262 @@
+
+Possible values are:
+* | ``"round"`` - Rounds the resulting size to the nearest integer value, with halfway cases rounded away from zero.
+* | ``"truncate"`` - Discards the fractional part of the resulting size.
+* | ``"ceil"`` - Rounds up the resulting size to the next integer value.)code",
+                                 "round")
+    .AddParent("ResizeAttrBase");
+
+
+void TensorResizeAttr::SetFlagsAndMode(const OpSpec &spec) {
+  has_sizes_ = spec.ArgumentDefined("sizes");
+  has_scales_ = spec.ArgumentDefined("scales");
+  has_max_size_ = spec.ArgumentDefined("max_size");
+  has_mode_ = spec.ArgumentDefined("mode");
+  has_axes_ = spec.ArgumentDefined("axes");
+  has_alignment_ = spec.ArgumentDefined("alignment");
+  subpixel_scale_ = spec.GetArgument<bool>("subpixel_scale");
+
+  DALI_ENFORCE(has_scales_ + has_sizes_ == 1, "Need one of ``scales`` or ``sizes``, but not both");
+
+  bool has_roi_start = spec.ArgumentDefined("roi_start");
+  bool has_roi_end = spec.ArgumentDefined("roi_end");
+  DALI_ENFORCE(has_roi_start == has_roi_end,
+               "``roi_start`` and ``roi_end`` must be specified together");
+  has_roi_ = has_roi_start && has_roi_end;
+  roi_relative_ = spec.GetArgument<bool>("roi_relative");
+
+  if (has_mode_) {
+    mode_ = ParseResizeMode(spec.GetArgument<std::string>("mode"));
+  } else {
+    mode_ = ResizeMode::Default;
+  }
+
+  if (has_axes_)
+    axes_ = spec.GetRepeatedArgument<int>("axes");
+
+  DALI_ENFORCE((mode_ != ResizeMode::NotSmaller && mode_ != ResizeMode::NotLarger) || !has_scales_,
+               "Providing ``scales`` is incompatible with not-smaller or not-larger modes");
+}
+
+template <typename T>
+void GetShapeLikeWithAxes(std::vector<T> &full, std::vector<T> &arg,
+                          const OpSpec& spec, const ArgumentWorkspace &ws,
+                          const std::string &arg_name, span<const int> axes,
+                          int nsamples, int spatial_ndim, int nargs,
+                          int first_spatial_dim) {
+  GetShapeLikeArgument<T>(arg, spec, arg_name, ws, nsamples, nargs);
+  full.resize(nsamples * spatial_ndim);
+  for (int sample_idx = 0; sample_idx < nsamples; sample_idx++) {
+    auto orig_arg = make_span(arg.data() + sample_idx * nargs, nargs);
+    auto full_arg = make_span(full.data() + sample_idx * spatial_ndim, spatial_ndim);
+    for (int i = 0; i < nargs; i++) {
+      int d = axes[i] - first_spatial_dim;
+      assert(d >= 0);
+      assert(d < spatial_ndim);
+      full_arg[d] = orig_arg[i];
+    }
+  }
+}
+
+void TensorResizeAttr::PrepareResizeParams(const OpSpec &spec, const ArgumentWorkspace &ws,
+                                           const TensorListShape<> &input_shape) {
+  SetFlagsAndMode(spec);
+
+  ndim_ = input_shape.sample_dim();
+  if (ndim_ < 1)
+    throw std::invalid_argument("Expected at least 1D inputs");
+  if (!has_axes_) {
+    axes_.resize(ndim_);
+    std::iota(axes_.begin(), axes_.end(), 0);
+  }
```

Comment:
nitpick: I understand it has much to do with the fact that the params depend on one another but it is quite hard to follow what is already loaded, needs to be defaulted and if it is validated if user provided.

Reply:
Do you have any suggestion about how to make it clearer?